tests/test_requests.py

```diff
@@ -482,6 +482,51 @@ def test_POSTBIN_GET_POST_FILES(self, httpbin):
         with pytest.raises(ValueError):
             requests.post(url, files=['bad file data'])
 
+    def test_POSTBIN_SEEKED_OBJECT_WITH_NO_ITER(self, httpbin):
+
+        class BufferedStream(object):
+            def __init__(self, data):
+                if isinstance(data, buffer):
+                    self.data = data
+                else:
+                    self.data = buffer(data)
+                self.length = len(self.data)
+                self.index = 0
+
+            def __len__(self):
+                return self.length
+
+            def read(self, size=None):
+                if size:
+                    ret = buffer(self.data, self.index, size)
+                    self.index += size
+                else:
+                    ret = buffer(self.data, self.index)
+                    self.index = self.length
+                return ret
+
+            def tell(self):
+                return self.index
+
+            def seek(self, offset, where=0):
+                if where == 0:
+                    self.index = offset
+                elif where == 1:
+                    self.index += offset
+                elif where == 2:
+                    self.index = self.length + offset
```

Comment:
Again, we only seek one kind of way here so we can avoid the over-generality I think. We can probably safely assert that `where == 0` and then just act like we're in that case.